player/screenshot.c

```diff
@@ -127,6 +127,30 @@ static void append_filename(char **s, const char *f)
     talloc_free(append);
 }
 
+static void trim_invalid_utf8(char *s, size_t len)
+{
+    for (int i = 1; i <= 3; i++) {
+        if (bstr_parse_utf8_code_length(s[len-i]) > i) {
+            s[len-i] = 0;
+            return;
+        }
+    }
+}
+
+// Long screenshot names (especially with %{sub-text}) can be longer than file
+// systems support. This helps and also removes trailing invalid codepoints.
```

Comment:
this should explain what the function does, namely trim the basename of a path so it is not longer than `max_utf8_bytes`

Reply:
I think it's fine in the latest push